src/launcher/plugin.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+  JSONObject
+} from 'phosphor/lib/algorithm/json';
+
+import {
+  JupyterLab, JupyterLabPlugin
+} from '../application';
+
+import {
+  ICommandPalette
+} from '../commandpalette';
+
+import {
+  IPathTracker
+} from '../filebrowser';
+
+import {
+  IServiceManager
+} from '../services';
+
+import {
+  ILauncher, LauncherModel, LauncherWidget
+} from './';
+
+
+/**
+ * A service providing an interface to the the launcher.
+ */
+export
+const launcherExtension: JupyterLabPlugin<ILauncher> = {
+  id: 'jupyter.extensions.launcher',
+  requires: [IServiceManager, IPathTracker, ICommandPalette],
+  provides: ILauncher,
+  activate: activateLauncher,
+  autoStart: true
+};
+
+
+/**
+ * Activate the launcher.
+ */
+function activateLauncher(app: JupyterLab, services: IServiceManager, pathTracker: IPathTracker, palette: ICommandPalette): ILauncher {
+  let launcherModel = new LauncherModel();
+
+  launcherModel.path = pathTracker.path;
+  launcherModel.app = app;
+
+  pathTracker.pathChanged.connect(() => {
+    launcherModel.path = pathTracker.path;
+  });
+
+  let launcherWidget = new LauncherWidget();
+
+  launcherWidget.model = launcherModel;
+  launcherWidget.id = 'landing-jupyterlab-widget';
+  launcherWidget.title.label = 'Launcher';
+
+  // Hardcoded defaults.
```

Comment:
I think we should add the one or two lines to each plugin (notebook, console, terminal, editor) to register these, instead of having a hardcoded default.